tests/test_upstream_versions.py

```diff
@@ -0,0 +1,162 @@
+      },
+      "_npmUser": {
+        "email": "benjaminrk@gmail.com",
+        "name": "minrk"
+      },
+      "_npmVersion": "5.4.2",
+      "author": {
+        "name": "Jupyter Developers"
+      },
+      "bin": {
+        "configurable-http-proxy": "bin/configurable-http-proxy"
+      },
+      "bugs": {
+        "url": "https://github.com/jupyterhub/configurable-http-proxy/issues"
+      },
+      "dependencies": {
+        "commander": "~2.13.0",
+        "http-proxy": "~1.16.2",
+        "lynx": "^0.2.0",
+        "strftime": "~0.10.0",
+        "winston": "~2.4.0"
+      },
+      "description": "A configurable-on-the-fly HTTP Proxy",
+      "devDependencies": {
+        "jasmine": "^2.5.1",
+        "jshint": "^2.9.2",
+        "nyc": "^11.0.2",
+        "prettier": "^1.4.4",
+        "request": "^2.81.0",
+        "request-promise-native": "^1.0.4",
+        "ws": "^4.0.0"
+      },
+      "directories": {
+      },
+      "dist": {
+        "integrity": "sha512-e+fxBy5cCayuNpxt3tcigBIuFsU/+oN48eK3aQtCBV12glavbBMxJa3ut2AEDHhXa/g3pC8r2BorKthrofHGRw==",
+        "shasum": "bc386574c519efeb8437960234d598be8fdd2030",
+        "tarball": "https://registry.npmjs.org/configurable-http-proxy/-/configurable-http-proxy-3.1.1.tgz"
+      },
+      "engineStrict": true,
+      "engines": {
+        "node": ">= 4.0"
+      },
+      "files": [
+        "index.js",
+        "lib/configproxy.js",
+        "lib/store.js",
+        "lib/trie.js",
+        "lib/error/*.html",
+        "bin/configurable-http-proxy"
+      ],
+      "gitHead": "91eb3e6c0e21a23d0c34277df42dfcbe71d52b8c",
+      "homepage": "https://github.com/jupyterhub/configurable-http-proxy#readme",
+      "license": "BSD-3-Clause",
+      "main": "index.js",
+      "maintainers": [
+        {
+          "email": "rgbkrk@gmail.com",
+          "name": "rgbkrk"
+        },
+        {
+          "email": "benjaminrk@gmail.com",
+          "name": "minrk"
+        }
+      ],
+      "name": "configurable-http-proxy",
+      "repository": {
+        "type": "git",
+        "url": "git+https://github.com/jupyterhub/configurable-http-proxy.git"
+      },
+      "scripts": {
+        "codecov": "nyc report --reporter=lcov && codecov",
+        "coverage-html": "nyc report --reporter=html",
+        "fmt": "prettier --write *.js bin/* lib/*.js test/*.js --trailing-comma es5 --print-width 100",
+        "lint": "jshint bin/ lib/ test/",
+        "test": "nyc jasmine JASMINE_CONFIG_PATH=test/jasmine.json"
+      },
+      "version": "3.1.1"
+    }
+  }
+}
+"""
+
+latest_url_test_list = [
+    (
+        sample_npm, "3.1.1", NPM(), {
+            "https://registry.npmjs.org/configurable-http-proxy": sample_npm_response
+        }
+    ),
+]
+
+
+@pytest.mark.parametrize("inp, ver, source, urls", latest_url_test_list)
+def test_latest_version(inp, ver, source, urls, requests_mock):
+    pmy = parse_meta_yaml(inp)["source"]
```

Comment:
What's the right way to transform the package into the thing that `get_latest_version` is expecting?

Reply:
You could use `get_attrs` https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/make_graph.py#L24 (although this currently pulls the `meta.yaml` itself from github. I guess at some point we should separate the request from the rest of it for these kinds of things.

Reply:
Hm, guess I can mock that, too...